clang/test/CodeGenHLSL/builtins/reversebits.hlsl

```diff
@@ -54,31 +29,6 @@ uint16_t4 test_bitreverse_ushort4(uint16_t4 p0)
 }
 #endif
 
-// CHECK: define noundef i32 @
-// CHECK: call i32 @llvm.bitreverse.i32(
-int test_bitreverse_int(int p0)
-{
-	return reversebits(p0);
-}

```

Comment:
Will these implicitly convert to `uint` now, or are these errors?

Reply:
errors specifically
`error: call to 'reversebits' is ambiguous`

because we can't distinguish int16_t, from int from int64_t to know which unsigned api to use.

Reply:
In that case we might need to document this as a difference from dxc's behaviour, which doesn't seem to care whether it's given a signed or unsigned int: https://hlsl.godbolt.org/z/xqjvzas1K

TBH I'm a bit worried this will come up a fair amount on existing shaders.

Reply:
Alright, let's get this in as is for now and loop back once the conversion story is in better shape.